tests/sles4sap/publiccloud/check_ibsm_embargoed.pm

```diff
@@ -0,0 +1,40 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+# Summary: Checks for embargoed updates on IBSM
+# Maintainer: QE-SAP <qe-sap@suse.de>
+
+use strict;
+use warnings;
+use base 'sles4sap_publiccloud_basetest';
+use publiccloud::utils "validate_repo";
+use testapi;
+
+sub test_flags {
+    return {fatal => 1, publiccloud_multi_module => 1};
+}
+
+sub run() {
+    my ($self, $run_args) = @_;
+    my $instance = $run_args->{my_instance};
+    $self->{network_peering_present} = 1 if ($run_args->{network_peering_present});
+    record_info('CONTEXT LOG', "instance:$instance network_peering_present:$self->{network_peering_present}");
+
+    my @repos = split(/,/, get_var('INCIDENT_REPO'));
+    my $count = 0;
+
+    while (defined(my $maintrepo = shift @repos)) {
```

Comment:
Why not `foreach my $maintrepo (@repos) {` ?

Reply:
I took this from `add_update_test_repo` as it was, but I can change it if you prefer this one

Reply:
I don't know. I suppose it works, but it looks all convoluted :D 

Up to you.

Reply:
I'll keep it for now, just to be on par with the rest of the peering/repo adding modules that use it. But it would be a matter of minutes to replace them all, so I propose a new ticket with an estimation of 1h to do that